File: src/main/java/net/insprill/cjm/messages/MessageVisibility.java
package net.insprill.cjm.messages;

public enum MessageVisibility {

    PUBLIC("Public"),
    PRIVATE("Private");

    private final String configSection;

    MessageVisibility(String configSection) {
        this.configSection = configSection;
    }

    public String getConfigSection() {
        return configSection;
    }

}

File: src/main/kotlin/net/insprill/cjm/message/MessageVisibility.kt
package net.insprill.cjm.message

enum class MessageVisibility(val configSection: String, private vararg val unsupportedActions: MessageAction) {
    PUBLIC("Public"),
    PRIVATE("Private", MessageAction.QUIT),
    ;

    fun supports(action: MessageAction): Boolean {
        return !unsupportedActions.contains(action)
    }

}
